client/landing/stepper/declarative-flow/internals/steps-repository/site-creation-step/index.tsx

```diff
@@ -88,13 +89,32 @@ const SiteCreationStep: Step = function SiteCreationStep( { navigation, flow } )
 		};
 	}
 
+	const [ freeFlowDomainCartItemLoaded, setFreeFlowDomainCartItemLoaded ] = useState( false );
+
 	useEffect( () => {
-		if ( submit ) {
+		if ( flow === 'free' ) {
+			// We hydrate the redux store with a dynamically generated domain when this view is first loaded
+			// because the free flow has no domains step.
+			const domainCartItem = domainRegistration( {
+				// TODO: Insert dynamically generated Domain Here
+				domain: '',
```

Comment:
**Note:** As noted by the inline comment, whenever the endpoint for dynamic generation of domain names is ready, we can fetch the information and use it here